kernel/kernel-api/src/main/java/io/delta/kernel/types/StructField.java

```diff
@@ -60,19 +63,36 @@ public class StructField {
   private final DataType dataType;
   private final boolean nullable;
   private final FieldMetadata metadata;
+  private final List<TypeChange> typeChanges;
 
   public StructField(String name, DataType dataType, boolean nullable) {
     this(name, dataType, nullable, FieldMetadata.empty());
   }
 
   public StructField(String name, DataType dataType, boolean nullable, FieldMetadata metadata) {
+    this(name, dataType, nullable, metadata, Collections.emptyList());
+  }
+
+  public StructField(
+      String name,
+      DataType dataType,
+      boolean nullable,
+      FieldMetadata metadata,
+      List<TypeChange> typeChanges) {
     this.name = name;
     this.dataType = dataType;
     this.nullable = nullable;
+    this.typeChanges = typeChanges == null ? Collections.emptyList() : typeChanges;
```

Comment:
IMO we should discourage `null` _ever_ being passed in. I agree this is a safe check but ... would be nice to never have to think about `null`?

Reply:
+1 probably can either (1) requireNonNull if we want this to always be present but empty for no type widening or (2) make it optional